packages/react-native-sdk/src/components/participants/VideoRenderer.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  CallingState,
+  SfuModels,
+  StreamVideoParticipant,
+  VisibilityState,
+} from '@stream-io/video-client';
+import { useCall, useCallCallingState } from '@stream-io/video-react-bindings';
+import { useMediaStreamManagement } from '../../providers';
+import { Z_INDEX } from '../../constants';
+import { ParticipantVideoFallback as DefaultParticipantVideoFallback } from './ParticipantVideoFallback';
+
+/**
+ * Props to be passed for the VideoRenderer component.
+ */
+export type VideoRendererProps = Pick<
+  ParticipantViewProps,
+  'ParticipantVideoFallback'
+> & {
+  /**
+   * The video kind that will be displayed.
+   * @types `screen` or `video`
+   */
+  videoMode: ParticipantVideoType;
+  /**
+   * The participant whose info will be displayed.
+   */
+  participant: StreamVideoParticipant;
+  /**
+   * When set to false, the video stream will not be displayed even if it is available.
+   *
+   * @default false
+   */
+  muteVideo?: boolean;
+};
+
+/**
+ * Lower level component, that represents only the video part (wrapper around the WebRTC)
+ */
+export const VideoRenderer = ({
+  videoMode,
+  participant,
+  muteVideo,
+  ParticipantVideoFallback = DefaultParticipantVideoFallback,
+}: VideoRendererProps) => {
+  const call = useCall();
+  const callingState = useCallCallingState();
+  const pendingVideoLayoutRef = useRef<SfuModels.VideoDimension>();
+  const subscribedVideoLayoutRef = useRef<SfuModels.VideoDimension>();
+  const { isCameraOnFrontFacingMode } = useMediaStreamManagement();
+  const {
+    isLocalParticipant,
+    sessionId,
+    publishedTracks,
+    viewportVisibilityState,
+    videoStream,
+    screenShareStream,
+  } = participant;
+
+  const isScreenSharing = videoMode === 'screen';
+  const isPublishingVideoTrack = publishedTracks.includes(
+    isScreenSharing
+      ? SfuModels.TrackType.SCREEN_SHARE
+      : SfuModels.TrackType.VIDEO,
+  );
+  const hasJoinedCall = callingState === CallingState.JOINED;
+  const canShowVideo = !!videoStream && !muteVideo && isPublishingVideoTrack;
+  const videoStreamToRender = isScreenSharing ? screenShareStream : videoStream;
+  const mirror = isLocalParticipant && isCameraOnFrontFacingMode;
+
+  /**
+   * This effect updates the participant's viewportVisibilityState
+   * Additionally makes sure that when this view becomes visible again, the layout to subscribe is known
+   */
+  useEffect(() => {
+    if (!call) {
+      return;
+    }
+    if (!muteVideo) {
+      if (viewportVisibilityState !== VisibilityState.VISIBLE) {
+        call.state.updateParticipant(sessionId, (p) => ({
+          ...p,
+          viewportVisibilityState: VisibilityState.VISIBLE,
+        }));
+      }
+    } else {
+      if (viewportVisibilityState !== VisibilityState.INVISIBLE) {
+        call.state.updateParticipant(sessionId, (p) => ({
+          ...p,
+          viewportVisibilityState: VisibilityState.INVISIBLE,
+        }));
+      }
```

Comment:
this will create a bug.. we need to inform invisible state at all cases..

that's why the prop was called `isVisible` if it was reported as `isVisible=false` we updated participant to invisible here

with this change it wont happen anymore!

Reply:
what triggered the change to call it muteVideo?

Reply:
isVisible was not relevant to any of the platforms hence it was renamed to `muteVideo` as React has `muteAudio`.

Reply:
Also, isVisible is kinda confusing as: to which stream's visibility you are talking about, audio/video.